src/scripts/view/productDetails.ts

```diff
@@ -0,0 +1,38 @@
+import { ApiServices } from '../controller/apiServices';
+import { templates } from './template';
+
+export class ProductDetails {
+    private apiServices: ApiServices;
+    private urlParams: URLSearchParams;
+    private productId: string;
+    private productDetailContainer: HTMLElement;
+
+    constructor() {
+        this.apiServices = new ApiServices();
+        this.urlParams = new URLSearchParams(window.location.search);
+        this.productId = this.urlParams.get('id')!;
```

Comment:
This will return null in case if Id is not available.

We can verify that ID is available before moving ahead.
```suggestion
        this.productId = this.urlParams.get('id') || '';
        
       if(!this.productId){
       // handle error
       }
```

Reply:
Fixed (https://github.com/krushna-simform/product-management-system/pull/3/commits/90833198d8880cbd6702f0e439ad7a08e65a140a)

Reply:
I think we didn't check what happened in case if we encounter no `productId`?

Reply:
My bad, i will implement that.

Reply:
Added validation check for `productId` (https://github.com/krushna-simform/product-management-system/pull/3/commits/ead9ecf3d136405bef8d8439211c15d569bdd3e2)